visidata/modify.py

```diff
@@ -123,7 +125,7 @@ def deleteBy(sheet, func, commit=False):
             sheet.deleteSourceRow(r)
             ndeleted += 1
 
-    if not commit:
+    if undo:
```

Comment:
This change is because we want committed deletes to have an undo, at the time of `commit-sheet`. We do not want them to undo before they are actually removed from the sheet.

This existed in the first place because sometimes I want to run deleteBy without setting an undo for it. As an example, if I am using it to undo added rows, then I want to be able to call it without it setting an undo for the undo-ing of those added rows. >_>

Reply:
So commit=True means "remove these rows from the sheet, do not just colour them". Undo=True means, set an undo for the removal of these rows for the current command.